nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    public static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            CLIENT_SERVICE, READER_SERVICE, WRITER_SERVICE, SUBMISSION_SCRIPT
+    ));
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("original")
+                                                    .description("The original flowfile")
+                                                    .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+                                                    .description("Flow files that fail to interact with graph server")
+                                                    .build();
+    public static final Relationship REL_ERRORS = new Relationship.Builder().name("errors")
+                                                    .description("Flow files that error in the response from graph server")
+                                                    .build();
+    public static final Relationship REL_GRAPH = new Relationship.Builder().name("response")
+                                                    .description("The response object from the graph server")
+                                                    .build();
+
+    public static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS, REL_FAILURE, REL_ERRORS, REL_GRAPH
+    )));
+
+    public static final String RECORD_COUNT = "records.count";
+    public static final String GRAPH_OPERATION_TIME = "graph.operations.took";
+    private volatile RecordPathCache recordPathCache;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    private GraphClientService clientService;
+    private RecordReaderFactory recordReaderFactory;
+    private RecordSetWriterFactory recordSetWriterFactory;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        clientService = context.getProperty(CLIENT_SERVICE).asControllerService(GraphClientService.class);
+        recordReaderFactory = context.getProperty(READER_SERVICE).asControllerService(RecordReaderFactory.class);
+        recordSetWriterFactory = context.getProperty(WRITER_SERVICE).asControllerService(RecordSetWriterFactory.class);
+        recordPathCache = new RecordPathCache(100);
+    }
+
+    private List<Object> getRecordValue(ProcessContext context, PropertyDescriptor entry, Record record, FlowFile input){
+        String valueRecordPath = context.getProperty(entry.getName()).evaluateAttributeExpressions(input).getValue();
+        final RecordPath recordPath = recordPathCache.getCompiled(valueRecordPath);
```

Comment:
Would be a good idea to have the dynamic properties compiled once per call to onTrigger.

Reply:
Refactored to calculate the recordPath cache once per call of onTrigger.